### PR TITLE
Updates to graphql implementation to Apollo client

### PR DESCRIPTION
**Pull request by Numbers88s**
## Context

Updating our graphql frontend implementation to use Apollo Client. See #24478 

## Changes
- Updates the CodeIntelConfigurationPage and underlying components.

### CODE REVIEW

File: dev/site-config.json [hunk 1]
```diff
@@ -10,5 +10,6 @@
   "update.channel": "release",
   "experimentalFeatures": {},
   "disablePublicRepoRedirects": true,
-  "repoListUpdateInterval": 1
+  "repoListUpdateInterval": 1,
+  "codeIntelAutoIndexing.enabled": true
```
Comment (efritz) [inline dev/site-config.json:14]: I put this in my own dev-private clone so that only code intel has to run the executors, which are kind of large.

**Review by efritz — CHANGES_REQUESTED** [review]
Found a few issues playing with storybook and/or a local stack (looking only at the configuration list page at the global and repository scopes):

- I don't see any delete/edit buttons for policies I've just created (ensuring they aren't protected).
- There's _only_ a button to create a new policy on the global policies pane, even in the repository-specific page (in that page, global policies should be read-only and users should be able to make a new repository-specific policy).

File: dev/site-config.json (hunk 1 above)
Comment (Numbers88s) [reply dev/site-config.json]: Sounds good, I removed the `"codeIntelAutoIndexing.enabled":true` bit. 

I also updated the component, you should see the delete/edit buttons now. I got the logic backwards 😄 

w/r/t
> There's only a button to create a new policy on the global policies pane, even in the repository-specific page (in that page, global policies should be read-only and users should be able to make a new repository-specific policy).

Thanks for the explanation 👍  It really helped. I sorted it out, but please let me know if missed something.

Comment (efritz) [reply dev/site-config.json]: I'll run it again tonight and do another review pass 😃

**Review by efritz — CHANGES_REQUESTED** [review]
Looks better! Seems like there's a missing index configuration tab in the repository view:

Screen Shot 2021-09-13 at 8 33 30 PM

And after deleting a policy from the list page, that policy should be removed from the list. Otherwise there's no indication of anything having happened, and doing it again causes a console error as the policy isn't found the second time.

**Review by efritz — APPROVED** [review]
Everything works locally!

**Pull request merged**